sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/BulkWriter.scala

```diff
@@ -942,25 +987,40 @@ private class BulkWriter(container: CosmosAsyncContainer,
       )
     }
 
-    if (numberOfIntervalsWithIdenticalActiveOperationSnapshots.get >= BulkWriter.maxAllowedMinutesWithoutAnyProgress) {
+    val secondsWithoutProgress = numberOfIntervalsWithIdenticalActiveOperationSnapshots.get *
+      writeConfig.flushCloseIntervalInSeconds
+    val maxAllowedIntervalWithoutAnyProgressExceeded =
+      secondsWithoutProgress >= writeConfig.maxRetryNoProgressIntervalInSeconds ||
+        (commitAttempt == 1
+          && this.activeReadManyOperations.isEmpty
```

Comment:
should the checking here also include the pendingReadManyRetries?

Reply:
Fixed